src/components/AnnotationPopover/AnnotationPopover.js

```diff
@@ -0,0 +1,78 @@
+// @flow
+import React from 'react';
+import noop from 'lodash/noop';
+import Overlay from 'box-react-ui/lib/components/flyout/Overlay';
+import ApprovalCommentForm from '../../../third-party/components/ApprovalCommentForm';
+
+import Internationalize from '../Internationalize';
+import AnnotationList from '../AnnotationList';
+
+import './AnnotationPopover.scss';
+
+type Props = {
+    annotations: Annotations,
+    canAnnotate: boolean,
+    onDelete: Function,
+    onCancel: Function,
+    onCreate: Function,
+    isPending: boolean,
+    language?: string,
+    messages?: StringMap
+};
+
+type State = {
+    isInputOpen: boolean
+};
+
+const NULL_USER = {};
+const IS_EDITING_TRUE = true;
+
+class AnnotationPopover extends React.Component<Props, State> {
```

Comment:
Can this be a `PureComponent`?

Reply:
The new commit i'm adding to this cannot be a PureComponent because it manages the state of whether or not the input of the ApprovalCommentForm is open or not